commands/build.go

```diff
@@ -643,6 +649,77 @@ func parsePrintFunc(str string) (*build.PrintFunc, error) {
 	return f, nil
 }
 
+func addGitProvenance(labels []string, contextPath string, dockerfilePath string) ([]string, error) {
+	if v, ok := os.LookupEnv("BUILDX_GIT_INFO"); ok && contextPath != "" {
+		if len(labels) == 0 {
+			labels = make([]string, 0)
+		}
+
+		// figure out in which directory the git command needs to run
+		var wd string
+		cwd, _ := os.Getwd()
+		if !filepath.IsAbs(contextPath) {
+			wd, _ = filepath.Abs(filepath.Join(cwd, contextPath))
+		} else {
+			wd = contextPath
+		}
+
+		// obtain Git sha of current HEAD
+		cmd := exec.Command("git", "rev-parse", "HEAD")
+		cmd.Dir = wd
+		sha, err := cmd.Output()
+		if err != nil {
+			return nil, fmt.Errorf("error obtaining git head: %w", err)
+		}
+
+		// check if the current HEAD is clean
+		cmd = exec.Command("git", "diff", "--quiet", "--exit-code")
```

Comment:
We should try and avoid the git porcelain commands I think, and try and use the plumbing equivalent: https://stackoverflow.com/questions/2657935/checking-for-a-dirty-index-or-untracked-files-with-git.

I think we also should check for staged files? Not just the unstaged ones.

Reply:
https://github.com/docker/buildx/blob/master/hack/dockerfiles/vendor.Dockerfile#L33 what we do for vendor checks. @corhere suggestions?

Reply:
Sorry, I'm confused a bit here: @jedevc you said to _avoid the git porcelain_ command but then @tonistiigi points at `git status --porcelain`. Not sure what to do ;-)

Reply:
Actually both `diff` and `status` like in the vendor script are wrong for this case. Diff commit as it is defined atm does not show any untacked(new) files are all. Status does that but still ignores the files in `.gitignore`. `git status --porcelain --ignored` seems to do the trick but there might be a better way.

Reply:
`git status --porcelain --ignored` sounds good to me: from `git-status`:
> Give the output in an easy-to-parse format for scripts. This is similar to the short output, but will remain stable across Git versions and regardless of user configuration. See below for details.

Reply:
I've switched to that for now.